### PR TITLE
NEXUS-8907 - SSL Certificates Functional Tests

### PR DESCRIPTION
**Pull request by kellyrob99**
https://issues.sonatype.org/browse/NEXUS-8907
http://bamboo.s/browse/NX3-OSSF535

### CODE REVIEW

File: testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      pemLoadCQ = '>>nx-coreui-sslcertificate-add-from-pem button[action=load]',
+      pemFieldCQ = '>>nx-coreui-sslcertificate-add-from-pem #pem',
+      addToTrustStoreButtonCQ = '>>nx-coreui-sslcertificate-details-form button[action=add][disabled=false]';
+
+  t.describe('SSL Certificates', function(t) {
+    t.it('An admin can load a certificate by pasting a PEM', function(t) {
+
+      t.chain(
+          t.openPageAsAdmin('admin/security/sslcertificates'),
+          {waitForCQVisible: sslCertificateListCQ},
```
Comment (kellyrob99) [inline testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js]: I find this more readable than the pure map form we have tended to use so far, i.e. 
```
{waitFor: 'CQVisible', args: sslCertificateListCQ}
```
Agree or disagree?

File: testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+              '/uW7JeOnXxWD4VVAuvHJFOGmaPaMoepQh1W4ueAy2GT1D45+PFeXQ8Yd0OO10c+n',
+              'wkOCAIGTk0wExqk/I/ytX9QrXdklowikn5se',
+              '-----END CERTIFICATE-----'
+            ].join('\n'), target: pemFieldCQ
+          },
+          {click: pemLoadCQ},
+          function(next) {
+            t.waitForAnimations(next);
+          },
+          {waitForCQVisible: addToTrustStoreButtonCQ, desc: 'button is enabled to add to trust store'},
```
Comment (kellyrob99) [inline testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js]: I'd also suggest we refrain from commenting code in favor of http://bryntum.com/docs/siesta/#!/api/Siesta.Test.Action .desc property, which shows up nicely in the test output and making it self-documenting

File: testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          {waitForCQVisible: '>>nx-coreui-sslcertificate-details-panel', desc: 'details are loaded for inspection'},
+          function(next, detailsPanel) {
```
Comment (kellyrob99) [inline testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js]: We can also potentially simplify some of the lookups using this pattern, where the result of the previous ComponentQuery is passed to the next step, in this case we get an Array of matches. This replaces the pattern where we wait for an element and then look it up again immediately to confirm the internal data.
Thoughts?

File: testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js (hunk 1 above)
Comment (mcculls) [inline testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js]: I find both just as readable, no objection if you want to use the new version

File: testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js (hunk 2 above)
Comment (mcculls) [inline testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js]: good idea

File: testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js (hunk 3 above)
Comment (mcculls) [inline testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js]: makes sense to me

Comment (adreghiciu) [inline testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js]: +1

File: testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js (hunk 1 above)
Comment (adreghiciu) [inline testsuite/functional-testsuite/src/test/ft-resources/testsuite-oss/security/security_ssl_certificates.t.js]: Looks to me like I always have to look in some other place to get the CQ so I like the previous version

**Comment by adreghiciu** [comment]
+1

**Comment by mcculls** [comment]
+1

**Pull request merged**